### ISSUE TITLE
3.12 fails to build on M1 macbook - Modules/_ssl.cpython-312-darwin.so': No such file or directory

### ISSUE THREAD

**Issue by gaborbernat** [pyenv/pyenv#2805]
```
❯ env CONFIGURE_OPTS='--enable-optimizations' pyenv install 3.12.0
python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 13.6 using python-build 2.3.28)

Inspect or clean up the working tree at /var/folders/v0/l8q3ghks2gs5ns2_p63tyqh40000gq/T/python-build.20231002105901.37245
Results logged to /var/folders/v0/l8q3ghks2gs5ns2_p63tyqh40000gq/T/python-build.20231002105901.37245.log

Last 10 log lines:
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/termios.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/termios.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_posixshmem.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_posixshmem.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_multiprocessing.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_multiprocessing.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_ctypes.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_ctypes.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_curses.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_curses.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_curses_panel.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_curses_panel.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_sqlite3.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_sqlite3.cpython-312-darwin.so
/opt/homebrew/opt/coreutils/libexec/gnubin/install -c -m 755 Modules/_ssl.cpython-312-darwin.so /Users/bgabor8/.pyenv/versions/3.12.0/lib/python3.12/lib-dynload/_ssl.cpython-312-darwin.so
install: cannot stat 'Modules/_ssl.cpython-312-darwin.so': No such file or directory
```

On 3.11 M1 macbook.

UPDATE: https://github.com/pyenv/pyenv/issues/2805#issuecomment-1744491474

**Comment by ITProKyle** [comment]
Same issue on an M1 Pro Macbook Pro running macOS 14.0.

**Comment by gaborbernat** [comment]
Seems the current config picks 1.1 openssl if installed, and gets confused.

```
brew uninstall --ignore-dependencies openssl@1.1
env CONFIGURE_OPTS='--enable-optimizations' pyenv install 3.12.0
brew install openssl@1.1
```

Works as a workaround.

**Comment by rodrigogiraoserrao** [comment]
Same issue here (M1 Mac, 13.3.1), @gaborbernat's suggestion also fixed it.

**Comment by anton-petrov** [comment]
This is log from my machine (macOS 13.6, Apple M1)
```bash
❯ pyenv install 3.12.0
python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk
Installed Python-3.12.0 to /Users/anton/.pyenv/versions/3.12.0
```

**Comment by petschki** [comment]
Same error here on macOS Sonoma 14.0, Apple M1 ... suggestion above fixes it!

> ```
> brew uninstall --ignore-dependencies openssl@1.1
> env CONFIGURE_OPTS='--enable-optimizations' pyenv install 3.12.0
> brew install openssl@1.1
> ```

**Comment by anton-petrov** [comment]
Good to have this lines in your bashrc or zshrc
```
export CPPFLAGS="-I$(brew --prefix)/include"
export LDFLAGS="-L$(brew --prefix)/lib"
```
Also, run `brew link openssl@3`

**Comment by kozlek** [comment]
> Good to have this lines in your bashrc or zshrc
> 
> ```
> export CPPFLAGS="-I$(brew --prefix)/include"
> export LDFLAGS="-L$(brew --prefix)/lib"
> ```
> 
> Also, run `brew link openssl@3`

I had remaining path in CPPFLAGS & LDFLAGS pointing to openssl@1.1 in my env, so even if pyenv reported using openssl@3, the compiler used the wrong one...
After updating them accordingly, it works like a charm 🙂

Thanks @anton-petrov

**Comment by gaborbernat** [comment]
IMHO the build script should shield from such env vars set. Forcing openssl 3 is not good as the user might use other tools that use v1 and we shouldn't break that.

**Comment by gaborbernat** [comment]
IMHO the build script should shield from such env vars set. Forcing openssl 3 is not good as the user might use other tools that use v1 and we shouldn't break that.

**Comment by samdoran** [comment]
Another option is to set `PYTHON_BUILD_SKIP_HOMEBREW=1` and make sure there aren't any Homebrew paths in `LDFLAGS` or `CPPFLAGS`. That potentially takes longer to build, but removes Homebrew from the equation.

**Comment by anton-petrov** [comment]
> IMHO the build script should shield from such env vars set. Forcing openssl 3 is not good as the user might use other tools that use v1 and we shouldn't break that.

What do you mean by "forcing"? I have both versions of openssl (1 & 3) installed on my laptop (mac m1 pro) using homebrew and CPython 3.12 compiles without errors. Pyenv does not break anything and does not affect the operation of the system in any way.

> OpenSSL 1.1.1 reached EOL [today ](https://mta.openssl.org/pipermail/openssl-announce/2023-September/000274.html)and got a final release for the occasion. AFAICT **all the supported Python versions that are still releasing binaries have changed to v3 already**, so… we can declare success? :upside_down_face: Or are there any outstanding immediate pain points / regressions etc.? In any case, thanks to everyone who worked on making this happen!
https://discuss.python.org/t/our-future-with-openssl/21486/89

**Comment by anton-petrov** [comment]
@gaborbernat any way, if you want to improve the build process and somehow improve the behavior of pyenv in this particular case, please, make some improvements and create a PR :)

**Comment by mrbigbunbury** [comment]
The following worked for me:

Requirements:

- Install/update `ncurses`, `openssl@3` and `sqlite3` via homebrew.
- Uninstall `openssl@1.1` via homebrew.

Installation command (with loadable SQLite extensions, which we in our case need for [SpatiaLite](https://www.gaia-gis.it/fossil/libspatialite/index)):

```
PYTHON_CONFIGURE_OPTS="--enable-loadable-sqlite-extensions --enable-optimizations --with-lto=full" \
LDFLAGS="-L$(brew --prefix ncurses)/lib -L$(brew --prefix sqlite3)/lib" \
CPPFLAGS="-I$(brew --prefix ncurses)/include -I$(brew --prefix sqlite3)/include" \
pyenv install 3.12.0
```

Remark: `--with-lto` did not work for me without installing `ncurses` via homebrew and adding the corresponding `LDFLAGS` and `CPPFLAGS` options.

**Comment by venaturum** [comment]
I had the same issue as TNieuwdorp:
```
install: Modules/_tkinter.cpython-312-darwin.so: No such file or directory
make: *** [sharedinstall] Error 71
```

None of the suggested solutions worked for me until I installed xquartz:
```
brew install --cask xquartz
```

Then I was able to succeed with the solution in jenhsun's post above
```
env PYTHON_BUILD_SKIP_HOMEBREW=1 PYTHON_CONFIGURE_OPTS="--enable-framework --enable-optimizations" pyenv install 3.12.0
```

**Comment by native-api** [comment]
This looks like MacOS' antivirus falsely detecting these files. Similar problems were reported before. A common solution was to reinstall Homebrew, to clear the antivirus' saved heuristics on it.

https://github.com/pyenv/pyenv/issues/2623#issuecomment-1436006967:

> MacOS 12+ has a built-in antivirus called 'XProtect'.
> 
> See #2603 and #2392 on how to view its logs and possible workarounds.

**Comment by kycool** [comment]
```python
export PATH="/usr/local/opt/openssl@3/bin:$PATH"
export LDFLAGS="-L/usr/local/opt/openssl@3/lib"
export CPPFLAGS="-I/usr/local/opt/openssl@3/include"
```

then `pyenv install 3.12.0` success

**Comment by jmmarschke** [comment]
After much effort, with:
```
% brew list
==> Formulae
autoconf	ca-certificates	git-lfs		m4		ncurses		openssl@3	pkg-config	pyenv		readline	sqlite		tcl-tk		xz

==> Casks
xquartz
```

This worked for me:
```
PYTHON_CONFIGURE_OPTS="--enable-framework --enable-loadable-sqlite-extensions --enable-optimizations --with-lto=full"\
LDFLAGS="-L$(brew --prefix xz)/lib -L$(brew --prefix ncurses)/lib -L$(brew --prefix sqlite3)/lib"\
CPPFLAGS="-I$(brew --prefix xz)/include -I$(brew --prefix ncurses)/include -I$(brew --prefix sqlite3)/include"\
pyenv install 3.12.0

python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use ncurses from homebrew
python-build: use zlib from xcode sdk
Installed Python-3.12.0 to ...
```

**Comment by harmv** [comment]
```
ginstall: cannot stat 'Modules/_blake2.cpython-312-darwin.so': No such file or directory
```

Fixed here with:
```
brew install libb2
```

After that:  `pyenv install 3.12.2` just worked

(Sonoma 14.4 on M2)

**Comment by native-api** [comment]
Without diagnostic information as per the issue template, we cannot say anything specific.

**Comment by ZacButko** [comment]
> ```
> ginstall: cannot stat 'Modules/_blake2.cpython-312-darwin.so': No such file or directory
> ```
> 
> Fixed here with:
> 
> ```
> brew install libb2
> ```
> 
> After that: `pyenv install 3.12.2` just worked
> 
> (Sonoma 14.4 on M2)
> 
> In my case @gaborbernat's suggestion did _not_ fix it.

This one was the fix for me! But for the last step I had to specify `arch --arm64 pyenv install 3.12`

**Comment by davidjrice** [comment]
* Apple M2
* macOS: 14.4.1
* Xcode: 15.3 (15E204a)

I haven't tried from a completely fresh install but after installation of `rbenv` and `ruby-build` (for Ruby dev; yes, please no flamerino 😹)

That leaves `homebrew` with the following packages;

* autoconf
* libyaml
* openssl@3
* rbenv
* ruby-build
* m4
* pkg-config
* readline

```
pyenv install 3.12.0
python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 14.4.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/jw/k99yncgx4ys0b2bdv_yxlp_40000gn/T/python-build.20240405091839.30974
Results logged to /var/folders/jw/k99yncgx4ys0b2bdv_yxlp_40000gn/T/python-build.20240405091839.30974.log

Last 10 log lines:
      __locale_localeconv in _localemodule.o
      __locale_localeconv in _localemodule.o
      __locale_localeconv in _localemodule.o
      __locale_localeconv in _localemodule.o
  "_libintl_textdomain", referenced from:
      __locale_textdomain in _localemodule.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Programs/_freeze_module] Error 1
make: *** Waiting for unfinished jobs....
```

Which was resolved ✅ with `brew install gettext`

```
pyenv install 3.12.2
python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.2.tar.xz...
-> https://www.python.org/ftp/python/3.12.2/Python-3.12.2.tar.xz
Installing Python-3.12.2...
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 14.4.1 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/jw/k99yncgx4ys0b2bdv_yxlp_40000gn/T/python-build.20240405091935.39707
Results logged to /var/folders/jw/k99yncgx4ys0b2bdv_yxlp_40000gn/T/python-build.20240405091935.39707.log

Last 10 log lines:
…
ginstall: cannot stat 'Modules/_blake2.cpython-312-darwin.so': No such file or directory
make: *** [sharedinstall] Error 1
```

Which was resolved ✅ with @harmv's tip `brew install libb2`

So in summary, to get a happy `pyenv` build environment after installing `rbenv`

```
brew install gettext libb2
```

**QUESTION:** should these perhaps be added as build dependencies to the `pyenv` homebrew package?

**Comment by nolafs** [comment]
Still issues with latest Mac OS

```
pyenv install 3.12.0   
python-build: use openssl@3 from homebrew
python-build: use readline from homebrew
Downloading Python-3.12.0.tar.xz...
-> https://www.python.org/ftp/python/3.12.0/Python-3.12.0.tar.xz
Installing Python-3.12.0...
python-build: use tcl-tk from homebrew
python-build: use readline from homebrew
python-build: use zlib from xcode sdk

BUILD FAILED (OS X 14.4 using python-build 20180424)

Inspect or clean up the working tree at /var/folders/wf/f3s34dks6js77vgk83h3x67c0000gn/T/python-build.20240410115425.52585
Results logged to /var/folders/wf/f3s34dks6js77vgk83h3x67c0000gn/T/python-build.20240410115425.52585.log

Last 10 log lines:
      __locale_localeconv in _localemodule.o
      __locale_localeconv in _localemodule.o
      __locale_localeconv in _localemodule.o
      __locale_localeconv in _localemodule.o
  "_libintl_textdomain", referenced from:
      __locale_textdomain in _localemodule.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Programs/_freeze_module] Error 1
make: *** Waiting for unfinished jobs....
```

**Comment by native-api** [comment]
@nolafs This is a different issue, listed in common builfd problems: https://github.com/pyenv/pyenv/wiki/Common-build-problems#on-apple-silicon-when-building-for-arm64-a-dependency-is-present-in-x64-homebrew-but-not-arm64-homebrew

**Comment by husainzgh** [comment]
> I had the same issue as TNieuwdorp:
> 
> ```
> install: Modules/_tkinter.cpython-312-darwin.so: No such file or directory
> make: *** [sharedinstall] Error 71
> ```
> 
> None of the suggested solutions worked for me until I installed xquartz:
> 
> ```
> brew install --cask xquartz
> ```
> 
> Then I was able to succeed with the solution in jenhsun's post above
> 
> ```
> env PYTHON_BUILD_SKIP_HOMEBREW=1 PYTHON_CONFIGURE_OPTS="--enable-framework --enable-optimizations" pyenv install 3.12.0
> ```

This one worked for me. Thank you @venaturum

**Comment by jordangullen** [comment]
For me the solution for the error

```
install: Modules/_tkinter.cpython-312-darwin.so: No such file or directory
make: *** [sharedinstall] Error 71
```

was

```
brew install tcl-tk
```

**Comment by woodens** [comment]
1. Install Required Dependencies
Make sure you have the necessary dependencies installed on your system. For macOS, you can install them using brew:
```
brew install openssl readline sqlite3 xz zlib tcl-tk
```
2. Configure Environment Variables
Set environment variables to let the build process know where to find these libraries:

```
export LDFLAGS="-L$(brew --prefix)/opt/zlib/lib -L$(brew --prefix)/opt/sqlite/lib -L$(brew --prefix)/opt/tcl-tk/lib"
export CPPFLAGS="-I$(brew --prefix)/opt/zlib/include -I$(brew --prefix)/opt/sqlite/include -I$(brew --prefix)/opt/tcl-tk/include"
export PKG_CONFIG_PATH="$(brew --prefix)/opt/zlib/lib/pkgconfig:$(brew --prefix)/opt/sqlite/lib/pkgconfig:$(brew --prefix)/opt/tcl-tk/lib/pkgconfig"
```
3. Re-attempt Python Installation
Now, try reinstalling Python 3.12.4 with pyenv:

``` 
pyenv install 3.12.4
```
finally,you will find that Installed Python-3.12.4 to /Users/xxxx/.pyenv/versions/3.12.4

**Comment by msqar** [comment]
I tried with all the solutions shared here but still having issues. I just don't know what else to try, this is the error I'm seeing after running `pyenv install 3.10.4` 

```
Inspect or clean up the working tree at /var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100
Results logged to /var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100.log

Last 10 log lines:
  File "/private/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100/Python-3.10.4/Lib/ensurepip/__init__.py", line 277, in _main
    return _bootstrap(
  File "/private/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100/Python-3.10.4/Lib/ensurepip/__init__.py", line 193, in _bootstrap
    return _run_pip([*args, *_PACKAGE_NAMES], additional_paths)
  File "/private/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100/Python-3.10.4/Lib/ensurepip/__init__.py", line 93, in _run_pip
    return subprocess.run([sys.executable, '-W', 'ignore::DeprecationWarning',
  File "/private/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100/Python-3.10.4/Lib/subprocess.py", line 524, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/private/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100/Python-3.10.4/python.exe', '-W', 'ignore::DeprecationWarning', '-c', '\nimport runpy\nimport sys\nsys.path = [\'/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/tmps6cptv6s/setuptools-58.1.0-py3-none-any.whl\', \'/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/tmps6cptv6s/pip-22.0.4-py3-none-any.whl\'] + sys.path\nsys.argv[1:] = [\'install\', \'--no-cache-dir\', \'--no-index\', \'--find-links\', \'/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/tmps6cptv6s\', \'--root\', \'/\', \'--upgrade\', \'setuptools\', \'pip\']\nrunpy.run_module("pip", run_name="__main__", alter_sys=True)\n']' died with <Signals.SIGABRT: 6>.
make: *** [install] Error 1
```

Any ideas?

**Comment by native-api** [comment]
> subprocess.CalledProcessError: Command '['/private/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/python-build.20240916164933.52100/Python-3.10.4/python.exe', '-W', 'ignore::DeprecationWarning', '-c', '\nimport runpy\nimport sys\nsys.path = [\'/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/tmps6cptv6s/setuptools-58.1.0-py3-none-any.whl\', \'/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/tmps6cptv6s/pip-22.0.4-py3-none-any.whl\'] + sys.path\nsys.argv[1:] = [\'install\', \'--no-cache-dir\', \'--no-index\', \'--find-links\', \'/var/folders/1g/4sc4cg5j0216_r0d5t3n6tjc0000gn/T/tmps6cptv6s\', \'--root\', \'/\', \'--upgrade\', \'setuptools\', \'pip\']\nrunpy.run_module("pip", run_name="__main__", alter_sys=True)\n']' died with <Signals.SIGABRT: 6>.
make: *** [install] Error 1

[That's a different problem altogether. Usually happens after upgrading to a new major MacOS version.](https://github.com/pyenv/pyenv/issues/2707)

**Comment by Torrent007** [comment]
> I had the same issue as TNieuwdorp:
> 
> ```
> install: Modules/_tkinter.cpython-312-darwin.so: No such file or directory
> make: *** [sharedinstall] Error 71
> ```
> 
> None of the suggested solutions worked for me until I installed xquartz:
> 
> ```
> brew install --cask xquartz
> ```
> 
> Then I was able to succeed with the solution in jenhsun's post above
> 
> ```
> env PYTHON_BUILD_SKIP_HOMEBREW=1 PYTHON_CONFIGURE_OPTS="--enable-framework --enable-optimizations" pyenv install 3.12.0
> ```

Works for me, thanks ! No idea why, but it's working on MacOS 15 and MBP-M1 <3

**Comment by mvl22** [comment]
> ```
> brew uninstall --ignore-dependencies openssl@1.1
> env CONFIGURE_OPTS='--enable-optimizations' pyenv install 3.12.0
> brew install openssl@1.1
> ```

This workaround by @gaborbernat now fails, giving:

`Error: openssl@1.1 has been disabled because it is not supported upstream! It was disabled on 2024-10-24.`

**Comment by native-api** [comment]
> This workaround by @gaborbernat now fails, giving:
> 
> `Error: openssl@1.1 has been disabled because it is not supported upstream! It was disabled on 2024-10-24.`

This should not be an issue, you should be able to get rid of OpenSSL 1.1, rebuilding any versions that use it against OpenSSL 3.
OpenSSL 3 support was added as early as in 3.8.1 -- but it was not official until 3.12.0rc2 .

**Comment by Elon-Lee** [comment]
> > Good to have this lines in your bashrc or zshrc
> > ```
> > export CPPFLAGS="-I$(brew --prefix)/include"
> > export LDFLAGS="-L$(brew --prefix)/lib"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Also, run `brew link openssl@3`
> 
> I had remaining path in CPPFLAGS & LDFLAGS pointing to openssl@1.1 in my env, so even if pyenv reported using openssl@3, the compiler used the wrong one... After updating them accordingly, it works like a charm 🙂
> 
> Thanks @anton-petrov

it is worked for me , thanks!

**Comment by adigandhi** [comment]
Still have an issue on Mac Studio M1 Max, the suggestion fixed it

**Comment by tcjuan** [comment]
After all above suggestion, I still got openssl module missing.  copy error log to chatgtp and found out it was Postgresql setting DYLD_LIB compile with different openssl lib.   
try
export DYLD_LIBRARY_PATH=""
and finally, it works!

**Comment by dotdev01** [comment]
installing the following package via macports solved the issue for me (mbp 2019, intel, macOS 13.7.6) .

`→ sudo port install py312-openssl`

**Comment by christopinka** [comment]
Can someone summarize what's going on here? Still struggling with this.

M1, Sequoia

Getting flags sourcing correctly with zsh fixed this for me using the recommended flags.
